rbac/management/group/view.py

```diff
@@ -579,3 +568,43 @@ def validate_and_get_exclude_key(self, params):
                 exclude)
             raise serializers.ValidationError({key: _(message)})
         return exclude
+
+    def obtain_principals(self, request, group):
+        """Obtain principals based on request, supports exclusion."""
+        principals_from_params = self.filtered_principals(group, request)
+
+        exclude = self.validate_and_get_exclude_key(request.query_params)
+
+        if exclude == 'false':
+            page = self.paginate_queryset(principals_from_params)
+            serializer = PrincipalSerializer(page, many=True)
+        else:
+            # Get all the principals to be excluded
+            serializer = PrincipalSerializer(principals_from_params, many=True)
+
+        principal_data = serializer.data
+        if principal_data:
+            username_list = [principal['username'] for principal in principal_data]
+        else:
+            username_list = []
+
+        # Call BOP to get principals
+        proxy = PrincipalProxy()
+
+        if exclude == 'false':
+            resp = proxy.request_filtered_principals(username_list, request.user.account)
+        else:
+            resp = proxy.request_principals(request.user.account, TOTAL_PRINCIPALS)
```

Comment:
Instead of setting `TOTAL_PRINCIPALS` here to `1000`, maybe we should just use `StandardResultsSetPagination.max_limit`? We use that in a few other places.

Reply:
Good to know! Will change this